### PR TITLE
update the todo by adding the localstorage

### PR DESCRIPTION
**Pull request by harshalhonde21**
updated the project and save the items in localstorage review and please merge it

### CODE REVIEW

**Comment by harshalhonde21** [comment]
please review and merge it added the localstorage part here

File: react/src/machine-coding/todo-list/todo.jsx [hunk 1]
```diff
@@ -8,14 +7,25 @@ const Todo = () => {
...
   const [editInfo, setEditInfo] = useState(null);
 
-  const submitHandler = (event) => {
+
+  useEffect(() => {
+    const storedItems = localStorage.getItem("todoItems");
+    if (storedItems) {
+      setItems(JSON.parse(storedItems));
+    }
+  }, []);
```
Comment (arpansaha13) [inline react/src/machine-coding/todo-list/todo.jsx:16]: This one is good

File: react/src/machine-coding/todo-list/todo.jsx [hunk 2]
```diff
@@ -8,14 +7,25 @@ const Todo = () => {
...
+
+  useEffect(() => {
+    localStorage.setItem("todoItems", JSON.stringify(items));
+  }, [items]);
```
Comment (arpansaha13) [inline react/src/machine-coding/todo-list/todo.jsx:20]: Perhaps there is a bug in these lines. This effect will run when the component mounts. This means the stored values in the local storage will be overwritten by the initial empty list. Which means the stored items are deleted on mount.

To prevent this you would want to skip the first render in this effect.
Please fix this bug.

File: react/src/main.tsx
Comment (arpansaha13) [inline react/src/main.tsx:1]: Please undo the changes in this file

**Review by arpansaha13 — CHANGES_REQUESTED** [review]

**Comment by arpansaha13** [comment]
After doing these changes, please remove this `changes-requested` label

**Comment by harshalhonde21** [comment]
done changes you can review it

**Review by arpansaha13 — CHANGES_REQUESTED** [review]
The todo items are not getting added in the local storage properly. The most recent todo-item doesn't get stored.

I would say that the previous code was correct. You just need to wrap the second useEffect with the `isInitialMount`. Let the first useEffect be as it was before.

And also please test your changes in your local before pushing. Refresh the page and see if all items are retained in your local.
